packages/protocol-kit/src/utils/passkeys/PasskeySigner.ts

```diff
@@ -0,0 +1,238 @@
+import { ethers, AbstractSigner, Provider } from 'ethers'
+import { Buffer } from 'buffer'
+
+import { PasskeyCoordinates, passkeyArgType } from '../../types/passkeys'
+import { SafeWebAuthnSignerFactoryContractImplementationType } from '../../types/contracts'
+
+// Sepolia only
+const P256_VERIFIER_ADDRESS = '0xcA89CBa4813D5B40AeC6E57A30d0Eeb500d6531b' // FCLP256Verifier
+
+/**
+ * Represents a Signer that is created using a passkey.
+ * This class extends the AbstractSigner to implement signer functionalities.
+ *
+ * @extends {AbstractSigner}
+ */
+class PasskeySigner extends AbstractSigner {
+  /**
+   * The raw identifier of the passkey.
+   * see: https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredential/rawId
+   */
+  passkeyRawId: ArrayBuffer
+
+  /**
+   * Passkey Coordinates.
+   */
+  coordinates: PasskeyCoordinates
+
+  /**
+   * Safe WebAuthn signer factory Contract.
+   */
+  safeWebAuthnSignerFactoryContract: SafeWebAuthnSignerFactoryContractImplementationType
+
+  constructor(
+    passkeyRawId: ArrayBuffer,
+    coordinates: PasskeyCoordinates,
+    safeWebAuthnSignerFactoryContract: SafeWebAuthnSignerFactoryContractImplementationType,
```

Comment:
What about using the safeProvider as a param and retrieve the contract and external provider from there?

Reply:
I think that the best option is to avoid this, because the user can use the custom contracts and we have all this logic in the `Safe.init()` function.

So to avoid to include this `customContracts` logic inside of the SafeProvider, I think that we should keep this:

```typescript

    const safeWebAuthnSignerFactoryContract = await getSafeWebAuthnSignerFactoryContract({
      safeProvider,
      safeVersion: '1.4.1',
      customContracts
    })

    const passkeySigner = await PasskeySigner.init(
      signer,
      safeWebAuthnSignerFactoryContract,
      safeProvider.getExternalProvider()
    )
    
     new SafeProvider({
      provider,
      signer: passkeySigner
    })
```